src/core/container.jl

```diff
@@ -166,14 +173,16 @@ function Libtask.consume(pc :: ParticleContainer)
         elseif score == Val{:done}
             num_done += 1
         else
-            error("[consume]: error in running particle filter.")
+            error("[consume]: error in running particle filter: likelihood=$(score)")
```

Comment:
I don't really understand what the return value of consume is. It seems it is either `<: Real`, `Val{:done}` or `nothing`. Is that correct? If so we should think about refactoring this bit at some point as it will likely improve performance.

Reply:
It's either `<:Real` (https://github.com/TuringLang/Turing.jl/blob/master/src/inference/AdvancedSMC.jl#L261) or `Val{:done}` (as we dicussed above). I don't think there is `nothing` procuded.

Reply:
Need verfication from @yebai

Reply:
Kai is correct, it can only be `<:Real` or `Val{:done}`.

Reply:
Thanks for the clarification.